### PR TITLE
JUnit5 execution capability

### LINKED ISSUE #6: Enable JUnit 5 by default instead of JUnit 4

### PR DESCRIPTION
**Pull request by rwqwr**
Closes https://github.com/RedMadRobot/gradle-infrastructure/issues/6

### CODE REVIEW

**Comment by osipxd** [comment]
Thank you for your contribution! I'll review it soon

File: src/main/kotlin/Accessors.kt [hunk 1]
```diff
@@ -28,3 +29,7 @@ internal fun <T : BaseExtension> Project.android(configure: T.() -> Unit) {
 inline fun Project.kotlinCompile(crossinline configure: KotlinCompile.() -> Unit) {
     tasks.withType<KotlinCompile> { configure() }
 }
+
+inline fun Project.kotlinTest(crossinline configure: Test.() -> Unit) {
```
Comment (osipxd) [inline src/main/kotlin/Accessors.kt]: Should be `internal`

File: src/main/kotlin/extension/RedmadrobotExtension.kt [hunk 2]
```diff
@@ -29,9 +30,16 @@ open class RedmadrobotExtension(objects: ObjectFactory) {
     /** Settings for android modules. */
     val android: AndroidSettings = AndroidSettings()
```
Comment (osipxd) [inline src/main/kotlin/extension/RedmadrobotExtension.kt]: Rename AndroidSettings to AndroidOptions, please :)
To make it consistent with TestOptions

File: src/main/kotlin/extension/RedmadrobotExtension.kt [hunk 3]
```diff
@@ -29,9 +30,16 @@ open class RedmadrobotExtension(objects: ObjectFactory) {
...
 
+    /** Settings for JVM test task */
+    val testOptions: TestOptions = TestOptions()
```
Comment (osipxd) [inline src/main/kotlin/extension/RedmadrobotExtension.kt]: Maybe rename it?
```suggestion
    val test: TestOptions = TestOptions()
```

File: src/main/kotlin/BaseAndroidPlugin.kt [hunk 4]
```diff
@@ -2,11 +2,14 @@ package com.redmadrobot.build
...
 import com.android.build.gradle.BaseExtension
 import com.redmadrobot.build.extension.redmadrobot
+import groovy.lang.Closure
 import org.gradle.api.JavaVersion
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.tasks.TaskProvider
+import org.gradle.api.tasks.testing.Test
 import org.gradle.kotlin.dsl.apply
+import org.gradle.kotlin.dsl.closureOf
```
Comment (osipxd) [inline src/main/kotlin/BaseAndroidPlugin.kt]: Unused imports here

**Review by osipxd — CHANGES_REQUESTED** [review]
There are some minor comments but looks good overall!
Update this branch to the latest `main` and I'll merge it :)

**Comment by osipxd** [comment]
It would be good if you add your work e-mail to your GitHub account.

**Review by osipxd — COMMENTED** [review]
Code is OK. Can you add brief docs about tests configuration to README? To "Usage" section.

**Comment by rwqwr** [comment]
Yes, I can

**Review by osipxd — APPROVED** [review]

**Comment by osipxd** [comment]
Thank you for your contribution!

**Pull request merged**